### PR TITLE
Fix many "import *" that should be "import ="

### PR DESCRIPTION
**Pull request by ghost**
Now that `--esModuleInterop` is enabled we can't `import *` a function.

### CODE REVIEW

**Review by andrewiggins — APPROVED** [review]

File: types/api-error-handler/api-error-handler-tests.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import * as errorHandler from 'api-error-handler';
+import errorHandler = require('api-error-handler');
 import * as express from 'express';
```
Comment (pheromonez) [inline types/api-error-handler/api-error-handler-tests.ts]: I'm curious as to why `import * as express` is something valid (for example, in this file) but not in many of the other files

Comment (andy-ms) [reply types/api-error-handler/api-error-handler-tests.ts]: We still allow it if it isn't used as a function, which it isn't in this file.

**Comment by dolanmiu** [comment]
From reading this:

https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/

I can see from examples that express should be:

```
import express from "express";

let app = express();
```

But you have made it:

```
import express = require('express');
```

Am I missing something here?

**Comment by taoqf** [comment]
how can we import types defined in same namespace?
```ts
declare namespace f {
  interface I {
  }
}
declare function f(): void;
declare module 'f' {
 export = f;
}
```
should we now use
```ts
import { I } from 'f';
import f = require('f');
```
instead of
```ts
import * as f from 'f';

const i: f.I;
```

**Comment by andy-ms** [comment]
@taoqf You can still use `f.I` if you want to.

**Comment by tlaziuk** [comment]
@dolanmiu I think it is called _ImportEqualDeclaration_ https://github.com/Microsoft/TypeScript/pull/19675/files#diff-d5e5579c5e7148db7139664593f6f03bR45 whatsmore it is a valid codefix https://github.com/Microsoft/TypeScript/pull/19675/files#diff-fe5e05e619b3b305344b55868f9739ca
but you are right, it's syntax looks strange 😄

**Comment by DanielRosenwasser** [comment]
Either works fine; the point of the interop flag is that we can bring CommonJS as a default. (@weswigham)

**Comment by weswigham** [comment]
Yeah. ImportEqualsDeclarations we get to say have the same behavior in esmoduleinterop mode as without, so using them instead of a synthetic default import in that mode or a namespace import outside of that mode gives you consistent behavior in both.

**Pull request merged**